aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -413,6 +420,97 @@ std::tuple<Tensor&,Tensor&> solve_out(Tensor& solution, Tensor& lu, const Tensor
+    "infos dtype ", infos.scalar_type(), " does not match the expected dtype ", kInt);
+  TORCH_CHECK(result.scalar_type() == input.scalar_type(),
+    "result dtype ", result.scalar_type(), " does not match input dtype ", input.scalar_type());
+  TORCH_CHECK(input.scalar_type() == other.scalar_type(),
+    "input dtype ", input.scalar_type(), " does not match other dtype ", other.scalar_type());
+
+  TORCH_CHECK(input.dim() >= 2,
+           "input should have at least 2 dimensions, but has ", input.dim(), " dimensions instead");
+  TORCH_CHECK(other.dim() >= 1,
+           "other should have at least 1 dimension, but has ", other.dim(), " dimensions instead");
+
+  // NumPy works for 1-dimensional 'other' or batch of 1-dimensional tensors, we need to unsqueeze it
+  // because 2-dimensional tensors are expected in the implementation
+  auto expected_batched_rhs_shape = IntArrayRef(input.sizes().data(), input.dim()-1);  // A.shape[:-1]
+  bool is_rhs_broadcasted = other.dim() == 1 || (input.dim()-1 == other.dim() && other.sizes().equals(expected_batched_rhs_shape));
+  Tensor other_ = is_rhs_broadcasted ? other.unsqueeze(-1) : other;
+
+  // _linalg_broadcast_batch_dims also includes linearSolveCheckInputs
+  // it checks for squareness of 'input' and 'shape' compatibility of 'other' and 'input'
+  Tensor other_broadcasted, input_broadcasted;
+  std::tie(other_broadcasted, input_broadcasted) = _linalg_broadcast_batch_dims(other_, input, "linalg_solve");
+
+  // if result has no elements we can modify it
+  if (result.numel() == 0) {
+    at::native::resize_as_(result, other_broadcasted.transpose(-2, -1), MemoryFormat::Contiguous);
+    result.transpose_(-2, -1);
+  } else {
+    // Resize messes up the strides and we expect strictly column major order, so let's not use at::native::resize_output
+    TORCH_CHECK(result.sizes().equals(other_broadcasted.sizes()),
+      "result shape ", result.sizes(), " does not match broadcasted other shape ", other_broadcasted.sizes());
+  }
+
+  TORCH_CHECK(result.transpose(-2, -1).is_contiguous(), "result tensor must be in batched column major order (Fortran contiguous).");
+  result.copy_(other_broadcasted);
+
+  auto input_working_copy = cloneBatchedColumnMajor(input_broadcasted);
+  at::native::resize_output(infos, {std::max<int64_t>(1, batchCount(input_broadcasted))});
+  // if input is empty infos might not get filled; make sure infos doesn't contain garbage then
+  if (input.numel() == 0) {
+    infos.fill_(0);
+  }
+  result = at::_linalg_solve_out_helper_(result, input_working_copy, infos);
+
+  // NumPy works for 1-dimensional 'other', we need to squeeze the result in this case
+  if (is_rhs_broadcasted) {
+    result.squeeze_(-1);
+  }
+
+  return result;
+}
+
+// Solves a system of linear equations matmul(input, x) = other in-place
+Tensor& linalg_solve_out(Tensor& result, const Tensor& input, const Tensor& other) {
+  auto infos = at::empty({0}, input.options().dtype(kInt));
```

Comment:
Curious about handling of infos here after reviewing the CUDA section.

Should this keep infos on CPU through the entire process? Since infos isn't exposed to users  it seems like there's little risk of doing so. Looking forward to your thoughts on this, @IvanYashchuk.